packages/expo-modules-core/ios/Core/Modules/CoreModule.swift

```diff
@@ -21,7 +21,13 @@ internal final class CoreModule: Module {
     Function("getViewConfig") { (viewName: String) -> [String: Any]? in
       var validAttributes: [String: Any] = [:]
       var directEventTypes: [String: Any] = [:]
-      let moduleHolder = appContext?.moduleRegistry.get(moduleHolderForName: viewName)
+      let parsedViewName: String
+      if let appIdentifier = appContext?.appIdentifier, viewName.hasSuffix("_\(appIdentifier)") {
+        parsedViewName = String(viewName.dropLast("_\(appIdentifier)".count))
+      } else {
+        parsedViewName = viewName
+      }
```

Comment:
This syntax should work as of Swift 5.9
On the other side, consider moving it to a separate function.
```suggestion
      let parsedViewName = if let appIdentifier = appContext?.appIdentifier, viewName.hasSuffix("_\(appIdentifier)") {
        String(viewName.dropLast("_\(appIdentifier)".count))
      } else {
        viewName
      }
```